### PR TITLE
Create missing CA config maps on deployment controller

### PR DESCRIPTION
**Pull request by jpkrohling**
This PR changes the deployment controller so that it always attempt to create the configmaps in the namespace for the target deployments, as long as sidecar injection is desired for the deployment.

﻿Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by rubenvp8510** [comment]
LGTM

File: pkg/controller/deployment/deployment_controller.go [hunk 1]
```diff
@@ -231,3 +220,46 @@ func (r *ReconcileDeployment) syncOnJaegerChanges(event handler.MapObject) []rec
...
+func (r *ReconcileDeployment) reconcileConfigMap(ctx context.Context, cm *corev1.ConfigMap, dep *appsv1.Deployment) error {
+	tracer := global.TraceProvider().GetTracer(v1.ReconciliationTracer)
+	ctx, span := tracer.Start(ctx, "reconcileConfigMap")
+	defer span.End()
+
+	cm.Namespace = dep.Namespace
+	span.SetAttribute(key.String("name", cm.Name))
+	span.SetAttribute(key.String("namespace", cm.Namespace))
+
+	// Update the namespace to be the same as the Deployment being injected
```
Comment (objectiser) [inline pkg/controller/deployment/deployment_controller.go:255]: Should be above line 251.

**Review by objectiser — COMMENTED** [review]
Only one minor comment.

File: pkg/controller/deployment/deployment_controller.go (hunk 1 above)
Comment (jpkrohling) [reply pkg/controller/deployment/deployment_controller.go]: Done

**Review by kevinearls — APPROVED** [review]
LGTM

**Review by objectiser — APPROVED** [review]

**Review by rubenvp8510 — APPROVED** [review]

**Comment by rubenvp8510** [comment]
Why the change from patch to update?

File: pkg/controller/deployment/deployment_controller.go [hunk 2]
```diff
@@ -108,74 +109,61 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 	err = r.rClient.Get(ctx, types.NamespacedName{Name: request.Namespace}, ns)
```
Comment (objectiser) [inline pkg/controller/deployment/deployment_controller.go:109]: @jpkrohling  @rubenvp8510  What is the difference between the `r.rClient` and `r.client`? Seems both are used in this class?

**Pull request merged**